Signal/src/util/DisplayableTextFilter.swift

```diff
@@ -6,36 +6,31 @@ import Foundation
 
 @objc class DisplayableTextFilter: NSObject {
 
-    // don't bother filtering on small text, lest we inadvertently catch legitimate usage of rare code point stacking
-    let allowAnyTextLessThanByteSize: Int
+    let TAG = "[DisplayableTextFilter]"
 
-    convenience override init() {
-        self.init(allowAnyTextLessThanByteSize: 10000)
-    }
-
-    required init(allowAnyTextLessThanByteSize: Int) {
-        self.allowAnyTextLessThanByteSize = allowAnyTextLessThanByteSize
-    }
-
-    @objc(shouldPreventDisplayOfText:)
-    func shouldPreventDisplay(text: String?) -> Bool {
+    @objc
+    func displayableText(_ text: String?) -> String? {
         guard let text = text else {
-            return false
+            return nil
         }
 
-        let byteCount = text.lengthOfBytes(using: .utf8)
-
-        guard byteCount >= allowAnyTextLessThanByteSize else {
-            return false
+        if (self.hasExcessiveDiacriticals(text: text)) {
+            return text.folding(options: .diacriticInsensitive, locale: .current)
```

Comment:
So if any character is invalid, we discard all diacriticals in the input. Seems reasonable since this should only happen for degenerate input.